swarm/src/connection.rs

```diff
@@ -57,12 +57,16 @@ use std::collections::HashSet;
 use std::fmt::{Display, Formatter};
 use std::future::Future;
 use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
 use std::task::Waker;
 use std::time::Duration;
 use std::{fmt, io, mem, pin::Pin, task::Context, task::Poll};
 
 static NEXT_CONNECTION_ID: AtomicUsize = AtomicUsize::new(1);
 
+///  Counter of the number of active streams on a connection
+type ActiveStreamCounter = Arc<()>;
```

Comment:
I'd have a slight preference for making this a proper new-type actually. I think for people not so familiar with Rust, having `Arc<()>` flying around here is not very intuitive :)

Reply:
https://github.com/libp2p/rust-libp2p/pull/4595#discussion_r1367989313 should be already resolved. 
fixed 0c9e3ee

Reply:
Oh sorry, adding this 41662a4. I also refactor stream a bit.

Reply:
Nice! Well spotted!